spice/training/actions.py

```diff
@@ -516,20 +522,75 @@ def tokenize_function(examples):
 
         return tokenized_dataset
 
+    def _process_dataset(self, dataset, config, task):
+        # get image processor
+        LOGGGER.info("Loading image processor...")
+        if task in ["train", "test"]:
+            hf_model_repo_id = config["hfModelRepoId"]
+            hf_model_repo_revision = config["hfModelRepoRevision"]
+
+            image_processor = AutoImageProcessor.from_pretrained(
+                pretrained_model_name_or_path=hf_model_repo_id,
+                revision=hf_model_repo_revision,
+                trust_remote_code=True,
+            )
+        elif task == "verify":
+            training_round_number = config["roundNumber"]
+            hf_model_repo_id = config["trainingJob"]["hfModelRepoId"]
+            base_model_repo_id = config["trainingJob"]["baseModelRepoId"]
+            base_model_repo_revision = config["trainingJob"]["baseModelRepoRevision"]
+
+            image_processor = AutoImageProcessor.from_pretrained(
+                pretrained_model_name_or_path=hf_model_repo_id
+                if hf_model_repo_id and training_round_number > 1
+                else base_model_repo_id,
+                revision="main"  # uses main tokenizer
+                if hf_model_repo_id and training_round_number > 1
+                else base_model_repo_revision,
+                trust_remote_code=True,
+            )
+        else:
+            error_message = f"_process_dataset with task={task} does not exist!"
+            LOGGER.error(error_message)
+            raise ValueError(error_message)
+
+        normalize = Normalize(
+            mean=image_processor.image_mean, std=image_processor.image_std
+        )
+        size = (
+            image_processor.size["shortest_edge"]
+            if "shortest_edge" in image_processor.size
+            else (image_processor.size["height"], image_processor.size["width"])
+        )
+        _transforms = Compose([RandomResizedCrop(size), ToTensor(), normalize])
```

Comment:
if `_transforms` and `normalize` are only used in the `transforms` function should we move them into it?